### PR TITLE
fix(amplify_datastore): avoid tearing down iOS subscription after clear()

### LINKED ISSUE #395: DataStore - iOS - observe() stops working after clear()
Amplify SDK Version: 0.1.0
Test project can be found here: https://github.com/kjones/datastore_test/

iOS Only. `DataStore.observe()` no longer works after calling `DataStore.clear()`. It all works great until a call to `DataStore.clear()` is made. From that point on, model change events cease. It doesn't matter if I cancel the `observe()` subscription before calling `clear()` or not. Current test code waits until the "ready" event before reissuing the `observe()`.

**Expected behavior**
There should be some magical combination of SDK calls that will get the `observe()` events going again after calling `clear()`.

**Platform**
Amplify Flutter current supports iOS and Android.  This issue is reproducible in (check all that apply):
[] Android
[x] iOS

**Smartphone (please complete the following information):**
 - Device: iOS Simulator

### PR DESCRIPTION
**Pull request by Amplifiyer**
*Issue #, if available:* fixes #395

*Description of changes:* fix(amplify_datastore): avoid tearing down iOS subscription after clear()

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift [hunk 1]
```diff
@@ -280,6 +284,10 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
                         flutterResult: flutterResult)
                 case .success():
                     print("Successfully cleared the store")
+                    // iOS tears down the publisher after clear. Let's setup again.
+                    // See https://github.com/aws-amplify/amplify-flutter/issues/395q
```
Comment (lawmicha) [inline packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift:288]:
```suggestion
                    // See https://github.com/aws-amplify/amplify-flutter/issues/395
```

**Review by lawmicha — APPROVED** [review]

File: packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift (hunk 1 above)
Comment (Amplifiyer) [reply packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift:288]: oops. thanks!

**Comment by kjones** [comment]
Pulled this branch locally and it fixed my test app linked to in #395 so 👍🏻 👍🏻 from me.

File: packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift [hunk 2]
```diff
@@ -280,6 +284,10 @@ public class SwiftAmplifyDataStorePlugin: NSObject, FlutterPlugin {
                         flutterResult: flutterResult)
                 case .success():
                     print("Successfully cleared the store")
+                    // iOS tears down the publisher after clear. Let's setup again.
+                    // See https://github.com/aws-amplify/amplify-flutter/issues/395
```
Comment (haverchuck) [inline packages/amplify_datastore/ios/Classes/SwiftAmplifyDataStorePlugin.swift:288]: I like citing the GH issue in this way.

**Review by haverchuck — APPROVED** [review]

**Pull request merged**